osm2lanes-web/src/draw.rs

```diff
@@ -193,20 +117,113 @@ pub fn lanes<R: RenderContext>(
     Ok(())
 }
 
-pub fn draw_arrow<R: RenderContext>(
+// TODO: refactor
+#[allow(clippy::too_many_arguments)]
```

Comment:
Honestly the type system saves us here, just imagine this is Python

Reply:
yeah,  totally workable and maintainable, just a bit of a code smell though.